pyomo/contrib/mindtpy/iterate.py

```diff
@@ -343,7 +331,7 @@ def algorithm_should_terminate(solve_data, config, check_cycling):
                 except (ValueError, OverflowError):
                     upper_slack = -10
```

Comment:
This value seems to be very arbitrary, wouldn't it be better to attach it to one of the parameters of MindtPy?

Reply:
Can we set it to `upper_slack =  -2*config.ecp_tolerance`? Here, we just need a number less than `-config.ecp_tolerance`.